src/openms/source/ANALYSIS/MAPMATCHING/QTClusterFinder.cpp

```diff
@@ -346,106 +375,105 @@ namespace OpenMS
 
             // Get the coordinates of the current cluster
-            const Int x = (*cluster)->getXCoord(); 
-            const Int y = (*cluster)->getYCoord();
+            const Int x = cluster.getXCoord();
+            const Int y = cluster.getYCoord();
 
             ////////////////////////////////////////
             // Step 1: Iterate through all neighboring grid features and try to
             // add elements to the current cluster to replace the ones we just
             // removed
-            const OpenMS::GridFeature* center_feature = (*cluster)->getCenterPoint();
-            addClusterElements_(x, y, grid, (**cluster), center_feature);
+            const GridFeature* center_feature = cluster.getCenterPoint();
+
+            addClusterElements_(x, y, grid, cluster, center_feature);
+
+            // update the heap, because the quality has changed
+            cluster_heads.update_lazy(handles[*id_it]);
 
             ////////////////////////////////////////
             // Step 2: update element_mapping as the best feature for each
             // cluster may have changed
-            typedef OpenMSBoost::unordered_map<Size,
-                    std::vector<GridFeature*> > NeighborList;
-            NeighborList neigh = (*cluster)->getAllNeighbors();
-            for (NeighborList::iterator n_it = neigh.begin(); n_it != neigh.end(); ++n_it)
+
+            NeighborMap neigh = cluster.getAllNeighborsDirect();
+            for (NeighborMap::iterator n_it = neigh.begin(); n_it != neigh.end(); ++n_it)
             {
-              for (std::vector<GridFeature*>::iterator i_it =
-                  n_it->second.begin(); i_it != n_it->second.end(); ++i_it)
-              {
-                // remember for each feature (gridfeature) all the cluster
-                // elements it belongs to
-                tmp_element_mapping[*i_it].push_back(*cluster);
-              }
+              GridFeature* gf_ptr = n_it->second.second;
+              tmp_element_mapping[gf_ptr].insert(cluster.getId());
             }
           }
         }
       }
 
-      for (ElementMapping::iterator it = tmp_element_mapping.begin(); 
+      for (ElementMapping::iterator it = tmp_element_mapping.begin();
           it != tmp_element_mapping.end(); ++it )
       {
-        for (std::vector<QTCluster*>::iterator it2 = it->second.begin();
+        for (unordered_set<Size>::iterator it2 = it->second.begin();
             it2 != it->second.end(); ++it2)
         {
-          element_mapping[ it->first ].push_back(*it2);
+          element_mapping[ it->first ].insert(*it2);
         }
       }
     }
+
+    // remove the current best from the heap
+    cluster_heads.pop();
   }
 
   void QTClusterFinder::addClusterElements_(int x, int y, const Grid& grid, QTCluster& cluster,
     const OpenMS::GridFeature* center_feature)
   {
     cluster.initializeCluster();
 
-#ifdef DEBUG_QTCLUSTERFINDER
+    #ifdef DEBUG_QTCLUSTERFINDER
```

Comment:
Preprocessing directives are usually not indented. So, one can easilier spot them.

Reply:
Is this a coding convention? I think it is super ugly and it messes with some of the indentation features of my code editor.